TK-499 : Ajouter un drapeau rouge

2snufkin opened the pull request.

Commit:
WIP: red flag in the fiche echant

Commit:
ref flag icon for the table

Commit:
removing un needed import

HUET-Sesan commented on tumorotek-webapp/src/main/java/fr/aphp/tumorotek/action/echantillon/FicheEchantillonStatic.java:
```diff
@@ -282,6 +301,13 @@ public void setObject(final TKdataObject e){
+      // Récupère la liste des retours associés à l'échantillon ayant un impact (dégradation probable de la qualité du matériel).
+      List<Retour>retours = ManagerLocator.getRetourManager().findByObjectAndImpactManager(echantillon, true);
+      // Définit une info-bulle pour l'icône d'impact indiquant la présence de retours avec impact.
+      impactIcon.setTooltiptext(Labels.getLabel("Champ.Retour.Impact"));
```
C'est surprenant d'affecter le tooltiptext ici. En effet, c'est une caractéristique de l'icône qui ne dépend pas de l'objet sélectionné donc ce n'est pas la peine de setter cette valeur à chaque changement d'échantillon sélectionné. Il vaut donc mieux mettre cette ligne de code dans doAfterCompose avec l'initialisation de visible.
Sinon, remarque générale : il n'est pas nécessaire de mettre un commentaire pour chaque ligne de code. Quand le nom de la méthode est explicite comme ici par exemple, ça "complique" plutôt la lecture.

HUET-Sesan replied:
Il était demandé de déplacer la ligne dans le doAfterCompose ce qui permettait d'avoir au même endroit les gestion du tooltip et de la visibilité du bouton. De plus par héritage, cette gestion était automatiquement reportée dans les écrans de sérologie et gatsbi. 
Au lieu de cela, le tooltip a été mis dans le zul ce qui a nécessité de le recopier dans les 3 fichiers.
**Je préfère que ce soit ce qui était demandé qui soit fait : le fait d'avoir la modification à un endroit au lieu de 3 simplifie la maintenance**

HUET-Sesan commented on tumorotek-webapp/src/main/webapp/zuls/cession/retour/ListeRetour2.zul:
```diff
@@ -46,6 +46,7 @@
 				hflex="1"
 				>
 				<columns sizable="true">
+					<column width="40px" />
```
Pourquoi avoir mis une largeur de 40 alors que les autres colonnes de ce tableau contenant une icône font 30 px ?

2snufkin replied:
Action corrective : j'ai maintenant réglé la largeur à 30 px pour aligner avec les autres colonnes contenant des icônes. Les 40 px initiaux étaient destinés à offrir plus de padding, mais la cohérence est prioritaire dans ce cas.

HUET-Sesan replied:
OK

HUET-Sesan commented on tumorotek-webapp/src/main/webapp/zuls/echantillon/FicheEchantillonStatic.zul:
```diff
@@ -50,10 +50,13 @@
+							<label id="codeLabelEchan" sclass="formValue" value="@{fwinEchantillonStatic$composer.object.code}" />
+							<!-- Image représentant une icône de drapeau pour signaler un impact
+							 (dégradation probable de la qualité du matériel). -->
+							<image id="impactIcon" src="/images/icones/flag.png"/>
```
Entre le label et l'image il faudrait ajouter une balise <separator/> pour espacer un peu

HUET-Sesan replied:
Utilisation de l'attribut spacing sur hbox à la place de la balise "separator" entre les 2 élements comme demandé. OK

HUET-Sesan commented on tumorotek-webapp/src/main/webapp/zuls/echantillon/gatsbi/FicheEchantillonStaticGatsbi.zul:
```diff
@@ -50,8 +50,13 @@
+							<!-- Image représentant une icône de drapeau pour signaler un impact
+                                 (dégradation probable de la qualité du matériel). -->
+							<image id="impactIcon" src="/images/icones/flag.png"/>
+							</hbox>
```
ajouter un <separator/> comme demandé dans FicheEchantillonStatic.zul

HUET-Sesan replied:
OK

HUET-Sesan commented on tumorotek-webapp/src/main/webapp/zuls/echantillon/serotk/FicheEchantillonStaticSero.zul:
```diff
@@ -50,10 +50,15 @@
+						<!-- Image représentant une icône de drapeau pour signaler un impact
+                             (dégradation probable de la qualité du matériel). -->
+						<image id="impactIcon" src="/images/icones/flag.png"/>
+						</hbox>
```
ajouter un <separator/> comme demandé dans FicheEchantillonStatic.zul

HUET-Sesan replied:
OK

HUET-Sesan commented on tumorotek-webapp/src/main/webapp/zuls/cession/retour/ListeRetour2.zul:
```diff
@@ -56,8 +57,14 @@
-						<label id="viewObj" value="@{currRetour.dateSortie}" class="formLink" style="@{currRetour.linkStyle}" />
+						<!-- Une icône pour signaler un impact (dégradation probable de la qualité du matériel). -->
+						<image src="/images/icones/flag.png" visible="@{currRetour.retour.impact}"/>
+						<label id="viewObj" value="@{currRetour.dateSortie}" class="formLink"
```
Pour être cohérent avec les autres utilisations du drapeau rouge, il faudrait ajouter un tooltiptext dans la balise image

HUET-Sesan replied:
La demande d'ajout du tooltiptext n'a pas été pris en compte. Dans ce cas, comme il s'agit d'un composant réutilisé  et que la visibilité est gérée dans le zul, l'attribut tooltiptext peut également être géré dans le zul.

HUET-Sesan reviewed (changes requested):
Il manque la gestion du drapeau rouge **sur la fiche statique d'un produit dérivé**.
Je remets la remarque générale indiquée sur une partie du code : l'ajout d'un commentaire dans le code doit être réservé au cas où cela apporte quelque chose (nom de méthode peu explicite ou explication d'une règle de gestion un peu compliquée).

2snufkin committed:
tooltip and space

Commit:
TK-499: code review fix

Commit:
code review: "ajout du tooltiptext"

Commit:
code review: tooltip dans doAfterCompose

Commit:
adding this for uniformity

HUET-Sesan committed:
Amélioration du code

- suppression d'une requête en base inutile
- centralisation de tout ce qui concerne la liste des retours (évènements) au même endroit
